### ISSUE TITLE
Zone 2 Support

### ISSUE THREAD

**Issue by mbbeaubi** [ToddGreenfield/homebridge-onkyo#1]
Hi,

Thanks for working on this, it's the missing piece for my home automation I think. If you had zone 2 support it would be perfect.

Should be easy enough to add, I read over the eISCP docs, it should be just a matter of prefixing all the commands you send with either main: or zone2:. Then all you'd need to do is add some config so the user can specify the zone.

If you're not already working on it I'm happy to test and contribute a fix.

Thanks

**Comment by ToddGreenfield** [comment]
Hey- glad to see it is being used beyond me! While I do not (yet) use zone2, I was planning on adding support for that. I was deciding on what to do next - adding support for A/B speaker support and a few other features or doing a complete re-write to make this into a Platform instead of an accessory as well as consolidating all the duplicated code. 

I also had not decided how best to handle zone2 support. I am not fond of the idea of adding a (fake) switch outside of the primary stereo switch, and was thinking more a toggle like how mute works now, but for zone support. Still kicking that around. If you have suggestions or would like to contribute, please feel free to test and do a PR. I am going on vacation next week and won't have time to do the re-write anyway for a couple weeks now.

**Comment by mbbeaubi** [comment]
I was bored and implemented it. Please take a look at my PR. On my stereo at least, zone 2 is completely separate from the main zone, so having a separate switch makes sense to me. Note that zone 2 is a different concept from A/B speakers, that probably should just be a mute like toggle if you decide to implement it. Let me know if you have any questions.

To configure just create another accessory for the second zone, and add a new parameter zone with value zone2.

I've tested it and it's working good. Only thing is that for me at least is I just see a single on / off toggle for power for the two stereos. I don't have volume control, mute, or anything. Maybe this is because I'm using the Apple Home app and not something better? Anyways, it works reliably for my use case, I just need to toggle the stereo on when I start playback on my Sonos Connect.

**Comment by mbbeaubi** [comment]
I just tried my patch with the eve app and I can see the volume control and mute on the second zone and it works. Changing volume is extremely clunky though. It seems like it is just slowly sending volume up commands until it reaches the desired volume. Maybe I should experiment with different models, right now I'm using TX-NR609.

Anyways, for my usecase, it's fine as I use Sonos to control volume. I just need my stereo to turn on and select the correct input, which this does nicely.

**Comment by ToddGreenfield** [comment]
@mbbeaubi I just merged and tried the PR and what I am seeing is essentially a duplication of all commands. I added a 2nd accessory, added the zone config and set it to "zone2". I see the accessory in both homekit and Eve, but when I change anything (input, volume, mute) it just updates both zones (main+zone2). Granted, I don't use my zone2, but I am able to manually change the input on my receiver, but it is not reflected back in the plugin logs or thru Eve. 
Are you using main and zone2 together? Are you able to control each independently and see the changes in Eve?
